src/actions/VariantsAction.php

```diff
@@ -0,0 +1,34 @@
+<?php
+/**
+ * HiPanel core package
+ *
+ * @link      https://hipanel.com/
+ * @package   hipanel-core
+ * @license   BSD-3-Clause
+ * @copyright Copyright (c) 2014-2019, HiQDev (http://hiqdev.com/)
+ */
+
+namespace hipanel\actions;
+
+use Yii;
+
+class VariantsAction extends RenderAction
+{
+    public array $variants;
```

Comment:
Тип переменной `array`, но в конструкторе нет обязательной инициализации. Потенциальный `Fatal`
```suggestion
    public array $variants = [];
```

```suggestion
    public ?array $variants = null;
```